ui/apps/profile/src/components/pages/profile-engagement/interests-page.tsx

```diff
@@ -55,7 +108,22 @@ const InterestsPage: React.FC<RootComponentProps> = props => {
     profileDataReq.data,
   );
 
-  const handleTopicClick = (topic: string) => {
+  const successText = `${t('Successfully subscribed to interest')}!`;
+
+  const handleTopicClick = topic => {
+    if (!ownTopics.find(interest => interest.value === topic.value)) {
+      createInterest
+        .mutateAsync({
```

Comment:
Unless you need concurrent mutations  or you absolutely need to use Promise, I don't see the reason for using mutateAsync. You will need to manually catch for errors but with mutate this isn't necessary.  So why not add the following inside the `onSuccess` method of the create interest hook ? :)
```
 setFeedbackMessage(successText);
 setShowFeedback(true);
```